centrallix/expression/exp_functions.c

```diff
@@ -3614,6 +3615,280 @@ int exp_fn_nth(pExpression tree, pParamObjects objlist, pExpression i0, pExpress
     return 0;
     }
 
+int exp_fn_levenshtein(pExpression tree, pParamObjects objlist, pExpression i0, pExpression i1, pExpression i2)
+    {
+
+    if (!i0 || !i1)
+	{
+		mssError(1,"EXP","levenshtein() requires two parameters");
+		return -1;
+	}
+
+    if ((i0->Flags & EXPR_F_NULL) || (i1->Flags & EXPR_F_NULL))
+	{
+		tree->DataType = DATA_T_INTEGER;
+		tree->Flags |= EXPR_F_NULL;
+		return 0;
+	}
+
+    if ((i0->DataType != DATA_T_STRING) || (i1->DataType != DATA_T_STRING))
+	{
+		mssError(1,"EXP","levenshtein() requires two string parameters");
+		return -1;
+	}
+
+	// for all i and j, d[i,j] will hold the Levenshtein distance between
+	// the first i characters of s and the first j characters of t
+	int length1 = strlen(i0->String);
+	int length2 = strlen(i1->String);
+	int levMatrix[length1+1][length2+1];
```

Comment:
This will need to be reworked.  Due to limited stack space in the usermode multithreading model, this large (and/or compile time indeterminate size) of an automatic variable can't be accommodated.